frontend/fingertips-frontend/components/organisms/Heatmap/heatmapUtil.ts

```diff
@@ -30,17 +30,16 @@ function getValueForAreaCode(
   areaCode: string,
   indicatorRowData: IndicatorRowData
 ): number | null {
-  let result = null;
   for (const healthDataForArea of indicatorRowData.rowData) {
     if (healthDataForArea.areaCode === areaCode) {
       for (const dataPoint of healthDataForArea.healthData) {
         if (dataPoint.year === indicatorRowData.year) {
-          result = dataPoint.value;
-          break;
+          // Coerce undefined to null
+          return dataPoint.value !== undefined ? dataPoint.value : null;
```

Comment:
This could use the nullish coalescing operator too, I think:
```suggestion
          return dataPoint.value ?? null;
```

Reply:
Although I think the better option might be to update the return type of this function to allow `undefined` instead of `null`.

Reply:
Changed to return undefined and updated the tests to also expect 'undefined'